jax/core.py

```diff
@@ -713,8 +727,13 @@ def pp_eqn(eqn):
           >> pp(' ') >> pp(pp_vars(eqn.invars))) + pp_subexpr
 
 def pp_jaxpr(jaxpr):
+  if len(jaxpr.outvars) > 1:
```

Comment:
Since jaxprs always have multiple results, I think we shouldn't treat `len(jaxpr.outvars) == 1` any different for pretty-printing.

Reply:
I think that printing "(x,)" instead of "x" does not really add any readability value. Similarly, in the left-hand side of a "let" we do not write "let x, = ...". 

If you still think it is best to not simplify the printing, I will revert this.

Reply:
It communicates that the output of evaluating a jaxpr is always a list/tuple and needs to be unpacked, as in `singleton_output_array, = core.eval_jaxpr(jaxpr, ...)`.

We do write let bindings involving primitives with multiple outputs differently (as in, when  `Primitive.multiple_results` is true we write `let a b = ...` rather than `let a = `). For single-output primitives I don't think it would make sense to unpack as in `let a, = ...`.

It seems that to be consistent for we should either always print parens/brackets or never print parens/brackets; that is, we should write `(a,)` and `(a, b, c)` for jaxpr outvars, or we should write `a` and `a b c`. (The latter would be more consistent with how we handle let bindings.) IIUC the current if/else logic here is mixing the two.

WDYT?

Reply:
I think that most of the time people who print Jaxprs are doing so to read the semantics, and for that purpose simplifying the syntax is fine. After all, the equations are not printed as a list either. The developers who need to write code that processes the Jaxpr, need to understand the internal structure, which they would understand better from the code.

I personally think that `(a,)` is too noisy for somebody who just wants to understand the Jaxpr. If we want consistency, I prefer to drop the parentheses all-together. But I am concerned that if we start having constants in the outvars, commas will make that list easier to read.